qtfred/src/ui/widgets/sexp_tree.cpp

```diff
@@ -1132,27 +1132,66 @@ int sexp_tree::get_default_value(sexp_list_item* item, char* text_buf, int op, i
 		return 0;
 
 		// Goober5000 - special cases that used to be numbers but are now hybrids
-	case OPF_GAME_SND:
-		gamesnd_id sound_index;
+		case OPF_GAME_SND:
+		{
+			gamesnd_id sound_index;
 
-		if (Operators[op].value == OP_EXPLOSION_EFFECT) {
-			sound_index = GameSounds::SHIP_EXPLODE_1;
-		} else if (Operators[op].value == OP_WARP_EFFECT) {
-			sound_index = (i == 8) ? GameSounds::CAPITAL_WARP_IN : GameSounds::CAPITAL_WARP_OUT;
-		}
+			if (Operators[op].value == OP_EXPLOSION_EFFECT)
+			{
+				sound_index = GameSounds::SHIP_EXPLODE_1;
+			}
+			else if (Operators[op].value == OP_WARP_EFFECT)
+			{
+				sound_index = (i == 8) ? GameSounds::CAPITAL_WARP_IN : GameSounds::CAPITAL_WARP_OUT;
+			}
 
-		if (sound_index.isValid()) {
-			game_snd* snd = gamesnd_get_game_sound(sound_index);
-			if (can_construe_as_integer(snd->name.c_str())) {
-				item->set_data(snd->name.c_str(), (SEXPT_NUMBER | SEXPT_VALID));
-			} else {
-				item->set_data(snd->name.c_str(), (SEXPT_STRING | SEXPT_VALID));
+			if (sound_index.isValid())
+			{
+				game_snd* snd = gamesnd_get_game_sound(sound_index);
+				if (can_construe_as_integer(snd->name.c_str()))
+					item->set_data(snd->name.c_str(), (SEXPT_NUMBER | SEXPT_VALID));
+				else
+					item->set_data(snd->name.c_str(), (SEXPT_STRING | SEXPT_VALID));
+				return 0;
 			}
-			return 0;
+
+			// if no hardcoded default, just use the listing default
+			break;
 		}
 
-		// if no hardcoded default, just use the listing default
-		break;
+		// Goober5000 - ditto
+		case OPF_FIREBALL:
+		{
+			int fireball_index = -1;
+
+			if (Operators[op].value == OP_EXPLOSION_EFFECT)
+			{
+				fireball_index = FIREBALL_MEDIUM_EXPLOSION;
+			}
+			else if (Operators[op].value == OP_WARP_EFFECT)
+			{
+				fireball_index = FIREBALL_WARP;
+			}
+
+			if (fireball_index >= 0)
+			{
+				extern fireball_info Fireball_info[MAX_FIREBALL_TYPES];
```

Comment:
This here causes the linker issue. Since you declare the extern reference inside this function which belongs to a namespace the compiler thinks that the extern reference also belongs in that namespace.

Instead, you should just declare the array in the header. That will solve the linking issue.

Reply:
Okay, that makes sense.  I was trying to hide the array from places where it didn't need to be visible, but putting it in the header is both easier and what the project usually does.